include/valgrind.h.in

```diff
@@ -1143,10 +1143,10 @@ typedef
 #define __SPECIAL_INSTRUCTION_PREAMBLE                            \
             ".option push\n\t"                                    \
             ".option norvc\n\t"                                   \
-            "srli a2, a2, 3\n\t"                                  \
-            "srli a2, a2, 13\n\t"                                 \
-            "srli a2, a2, 51\n\t"                                 \
-            "srli a2, a2, 61\n\t"
+            "srli a0, a0, 3\n\t"                                  \
```

Comment:
This version corrupts a0 instead of a2. What I meant by updating it to zero is to change the preamble to use the zero/x0 register, sorry for not being clear.

Reply:
Ooh my bad mixed up the alias :/